Cesium.Parser/CParser.cs

```diff
@@ -158,13 +161,19 @@ public partial class CParser
 
     // TODO: [Rule("declaration_specifiers: storage_class_specifier declaration_specifiers?")]
     [Rule("declaration_specifiers: type_specifier declaration_specifiers?")]
-    // TODO: [Rule("declaration_specifiers: type_qualifier declaration_specifiers?")]
-    // TODO: [Rule("declaration_specifiers: function_specifier declaration_specifiers?")]
-    // TODO: [Rule("declaration_specifiers: alignment_specifier declaration_specifiers?")]
     private static DeclarationSpecifiers MakeDeclarationSpecifiers(
         TypeSpecifier typeSpecifier,
         DeclarationSpecifiers? rest) =>
-        rest?.Insert(0, typeSpecifier) ?? ImmutableArray.Create((DeclarationSpecifier)typeSpecifier);
+        rest?.Insert(0, typeSpecifier) ?? ImmutableArray.Create<IDeclarationSpecifier>(typeSpecifier);
+
+    [Rule("declaration_specifiers: type_qualifier declaration_specifiers?")]
+    private static DeclarationSpecifiers MakeDeclarationSpecifiers(
+        TypeQualifier typeQualifier,
+        DeclarationSpecifiers? rest) =>
+        rest?.Insert(0, typeQualifier) ?? ImmutableArray.Create<IDeclarationSpecifier>(typeQualifier);
+
+    // TODO: [Rule("declaration_specifiers: function_specifier declaration_specifiers?")]
```

Comment:
Shouldn't these TODO located near `MakeDeclarationSpecifiers`?

Reply:
The order of these TODOs strictly corresponds to the standard, for easier orientation.